src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArr = [];
+  let clonedState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use a destruction for action
```suggestion
  for (const action of actions) {
    const { type, extraData, keysToRemove} = action;
    
    switch (type) {
```